src/coord/src/coord.rs

```diff
@@ -7,33 +7,67 @@
 // by the Apache License, Version 2.0.
 
-//! Coordination of installed views, available timestamps, compacted timestamps, and transactions.
+//! Coordination of installed views, available timestamps, compacted
+//! timestamps, and transactions.
 //!
-//! The command coordinator maintains a view of the installed
-//! views, and for each tracks the frontier of available times
-//! ([`upper`](arrangement_state::Frontiers::upper)) and the frontier
-//! of compacted times ([`since`](arrangement_state::Frontiers::since)).
-//! The upper frontier describes times that may not return immediately, as any
-//! timestamps in advance of the frontier are still open. The since frontier
-//! constrains those times for which the maintained view will be correct,
-//! as any timestamps in advance of the frontier must accumulate to the same
-//! value as would an un-compacted trace. The since frontier cannot be directly
-//! mutated, but instead can have multiple handles to it which forward changes
-//! from an internal MutableAntichain to the since.
+//! The command coordinator maintains a view of the installed views, and for
+//! each tracks the frontier of available times
+//! ([`upper`](arrangement_state::Frontiers::upper)) and the frontier of
+//! compacted times ([`since`](arrangement_state::Frontiers::since)). The upper
+//! frontier describes times that may not return immediately, as any timestamps
+//! in advance of the frontier are still open. The since frontier constrains
+//! those times for which the maintained view will be correct, as any
+//! timestamps in advance of the frontier must accumulate to the same value as
+//! would an un-compacted trace. The since frontier cannot be directly mutated,
+//! but instead can have multiple handles to it which forward changes from an
+//! internal MutableAntichain to the since.
+//!
+//! The [`Coordinator`] tracks various compaction frontiers so that indexes,
+//! compaction, and transactions can work together.
+//! [`determine_timestamp()`](Coordinator::determine_timestamp) returns the
+//! least valid since of its sources. Any new transactions should thus always
+//! be >= the current compaction frontier and so should never change the
+//! frontier when being added to [`txn_reads`](Coordinator::txn_reads). The
+//! compaction frontier may change when a transaction ends (if it was the
+//! oldest transaction and the index's since was advanced after the transaction
+//! started) or when [`update_upper()`](Coordinator::update_upper) is run (if
+//! there are no in progress transactions before the new since). When it does,
+//! it is added to [`since_updates`](Coordinator::since_updates) and will be
+//! processed during the next [`maintenance()`](Coordinator::maintenance) call.
+//!
+//! ## Frontiers another way
+//!
+//! If the above description of frontiers left you with questions, this
+//! repackaged explanation might help.
+//!
+//! - `upper` is the most recent time that you can read from sources. Reads at
```

Comment:
This implies that it's impossible or incorrect to read at a time greater than upper, but that's not quite the case. The second sentence explains it correctly, so it might make sense to rejigger the first. I tried to come up with examples but they were all long wordy sentences that might defeat the purpose of this paragraph, so...maybe it's fine or needs more thought?